v2/cmd/controller/main.go

```diff
@@ -26,6 +26,8 @@ func main() {
 		os.Exit(1)
 	}
 
+	setupLog.Info("Launching with flags", "flags", flgs.String())
```

Comment:
Is this one of those loggers that accepts name/value pairs? if so, maybe we want to log the parameters individually.

Reply:
Done

Reply:
Actually I take it back: I tried to do this, but actually making it work with the logger is quite awkward.

You need a method like this:
```
func (f Flags) Values() []any {
	return []any{
		"MetricsAddr", f.MetricsAddr,
		"HealthAddr", f.HealthAddr,
		"EnableLeaderElection", "%t", f.EnableLeaderElection,
		"CRDPatterns", f.CRDPatterns,
		"PreUpgradeCheck", "%t", f.PreUpgradeCheck,
	}
}
```

and then you also _must_ use it like:
`setupLog.Info("Launching with flags", flgs.Values()...)`

if you miss the `...` trailing, there's a runtime panic.

Given the above, I've moved back to logging it as a string like we do for `config.Vars`. That's consistent and it avoids the risk of programmer mistake leading to panic.